scraping-formula1/scraping_formula1_solution.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+"""scraping-formula1_solution
+
+Automatically generated by Colab.
+
+Original file is located at
+    https://colab.research.google.com/drive/1NrElKaMCO7YlzM6L8pvnJ-XVbpISZrTS
+"""
+
+from bs4 import BeautifulSoup
+import pandas as pd
+import requests
+
+url = 'https://www.formula1.com/en/results/2022/drivers'
+
+r = requests.get(url)
+
+r.status_code # Success!
+
+data = BeautifulSoup(r.content)
+data
+
+data.find_all('table')
+
+table = data.table #table
+
+table_rows = table.find_all('tr') #table rows
+header_row = table_rows[0]
+data_rows = table_rows[1:]
+
+headers = header_row.find_all('th') #table headers
+colnames = [name.text for name in headers]
+colnames
+
+rows = []
+for row in data_rows:
+  tds = row.find_all('td')
+  tds_data = [td.text for td in tds]
+  rows.append(tds_data)
+
+drivers = pd.DataFrame(data = rows, columns = colnames)
```

Comment:
It's not necessary for the assignment, but there's some cleaning that can be done on the "Driver" column that might make it look nicer.